utils/keygen/mkkey.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2014 The Dename Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package main
+
+import (
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	"golang.org/x/crypto/ed25519"
+)
+
+const (
+	SECRET_KEY string = "ed25519.secret"
```

Comment:
Calling these "ed25519" goes against our 'hide the primitive' principal.  `sign.secret`, `sign.pub` would be better.